pkg/buses/gcppubsub/bus.go

```diff
@@ -70,15 +71,15 @@ func NewCloudPubSubBusDispatcher(ref buses.BusReference, projectID string, opts
 	return bus, nil
 }
 
-func NewCloudPubSubBusProvisioner(ref buses.BusReference, projectID string, opts *buses.BusOpts) (*CloudPubSubBus, error) {
+func NewCloudPubSubBusProvisioner(ref buses.BusReference, busUID string, projectID string, opts *buses.BusOpts) (*CloudPubSubBus, error) {
 	ctx := context.Background()
 	pubsubClient, err := pubsub.NewClient(ctx, projectID)
 	if err != nil {
 		return nil, err
 	}
 
 	bus := &CloudPubSubBus{
-		ref:          ref,
+		id:           busId(busUID),
```

Comment:
rather than storing the id like this I wonder if it might be cleaner to set a topic prefix field that is generated from the id and our special strings (like `channel-` and then `topicID` becomes

```
func (b *CloudPubSubBus) topicID(channel buses.ChannelReference) string {
	return fmt.Sprintf("%s-%s", b.topicPrefix, channel.Namespace, channel.Name)
}
```

And then we are free to change how we generate that prefix, and the code does not have to change each time.

Reply:
It seems like we *should* store the generated UID in the status so we can recover it in the future.

Additionally, storing the topic ID means that we don't get thrown off later if we change the `busID` function.

Since this is the old Bus code, it's probably not worth updating the `status` to support this, but we should probably include a `runtime.RawExtension` in the `status` of the new Channel to support recording controller choices like this.